User-Data: T2116: Added User-Data handler

zdc opened the pull request:
User-Data handler allows to:
* process configuration commands
* preload the whole configuration file
* install the system to permanent storage

All of this allows zero-touch provision with the full configuration.

zdc committed:
User-Data: T2116: Added User-Data handler

User-Data handler allows to:
* process configuration commands
* preload the whole configuration file
* install the system to permanent storage

All of this allows zero-touch provision with the full configuration.

ocanty commented:
what's the status on this? it'd be very useful if one could have one time config/setup when provisioning a vyos instance

zdc commented:
@ocanty in a testing stage. All announced functionality works well, but maybe we will change something after tests. If you want to participate in testing, feel free to contact us in Slack.

zdc committed:
User-Data: T2116: Added some tricks to pass build tests

We need these changes if we want to use and pass tests provided in upstream Cloud-init package during package build

zdc commented:
This User-Data handler was tested for some time in different platforms, and after receiving feedback we decided to keep this version in the format of an independent handler file (for including as part of User-Data itself) and create for integration a new one variant.
